### DISCUSSION TITLE
XGB predict_proba and predict method not defined self.X

### DISCUSSION THREAD

**Discussion by MatufA** [vertica/VerticaPy#1269] [Potential bug] [answered]
After training new XGBClassifier I tried to load and predict using `predict_proba` and `predict` methods and got the error:
```
AttributeError: 'XGBClassifier' object has no attribute 'X'
```
After debugging the source code I noticed they not initialized correctly the self.X attribute.
for example in `predict_proba` implementation:
https://github.com/vertica/VerticaPy/blob/4d842285f698a7b8fad2c4931dbeb67e3a78231a/verticapy/machine_learning/vertica/base.py#L4595-L4596

self.X never define before (same for predict).

the only time the methods work fine for me was after `fit`, due to initialization in fit I suspect.
https://github.com/vertica/VerticaPy/blob/4d842285f698a7b8fad2c4931dbeb67e3a78231a/verticapy/machine_learning/vertica/base.py#L2313-L2319

**Comment by oualib** [comment]
Thanks for reporting this issue, I'll do some tests and come back to you.

**Comment by oualib** [comment, marked as answer]
You can not predict if you did not train a model.

I just trained and load a ``XGBClassifier `` and I got no issue.

```python
from verticapy.machine_learning.vertica import XGBClassifier

table = "titanic"
X = ["fare", "age"]
y = "survived"
model = XGBClassifier("model_test_260924")
model.drop()
model.fit(table, X, y)
model.X
```

Result:
```python
['"fare"', '"age"']
```

Loading the model and checking attribute ``X``:
```python
from verticapy.machine_learning.vertica import load_model

model2 = load_model("model_test_260924")
model2.X
```

Result:
```python
['fare', 'age']
```

Besides, it is obvious that attributes X and y are created when fitting the model. It is always advise to write entirely the table name, X and y whenever making a prediction AS the variables name might not be the same in the new table.
